File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsCourseReviews.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class ReviewEligibility(
    val eligible: Boolean = false,
    val progressPercent: Int = 0,
    val hasReview: Boolean = false,
    val canEdit: Boolean = false,
    val reviewId: String? = null,
)

@Serializable
data class SubmitCourseReviewRequest(
    val rating: Int,
    val reviewText: String? = null,
)

@Serializable
data class SubmittedCourseReview(
    val id: String,
    val courseId: String,
    val reviewerId: String,
    val rating: Int,
    val reviewText: String? = null,
    val reviewerDisplayName: String,
    val createdAt: String,
    val updatedAt: String,
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsCourseReviews.swift
import Foundation

// MARK: - Course reviews submit (M9.3)

struct ReviewEligibility: Codable {
    var eligible: Bool
    var progressPercent: Int
    var hasReview: Bool
    var canEdit: Bool
    var reviewId: String?
}

struct SubmitCourseReviewRequest: Encodable {
    var rating: Int
    var reviewText: String?
}

struct SubmittedCourseReview: Codable, Identifiable, Hashable {
    var id: String
    var courseId: String
    var reviewerId: String
    var rating: Int
    var reviewText: String?
    var reviewerDisplayName: String
    var createdAt: String
    var updatedAt: String
}